src/main/webapp/app/shared/server-date.service.ts

```diff
@@ -24,11 +28,12 @@ export class ArtemisServerDateService {
                 }
             });
         } else {
-            // definitly sync if we do not have 5 elements yet
+            // definitely sync if we do not have 5 elements yet
             shouldSync = true;
         }
         // TODO: one additional optimization could be to take the duration for request -> response into account here
```

Comment:
Is anyone working on this? Otherwise, I might give this a shot! This can be a significant factor and make the accuracy worse than it actually is!

Reply:
@jpbernius I'm not working on this TODO right now and I guess nobody else is working on this in general. So feel free to give this a shot! 🚀

Reply:
Please talk to @stefanwaldhauser: he wrote a seminar paper about how this works best. It can become quite complex though. He told me you, that you also need to send the client time to the server and you need at least 3 calls.

We actually designed a new REST endpoint /time, which is without access control and database access, so it should be super fast (even with a bad internet connection it should not take longer than a few ms, you can test this with the Edge profile in Chrome), so I am not sure if it makes sense to implement additional logic now. I would prefer that we do this in a follow up PR.

We should first test that the implementation of this PR is correct (especially the logic with the offset) and that a wrong time zone / wrong local clock at the start of the exam still shows the correct working time. Then we should merge this PR and if we find time, we can improve this in a follow up.

Reply:
OK, so we keep it as is for now. If we need additional logic, we can consider integrating the [timesync](https://www.npmjs.com/package/timesync) library.